README.md

```diff
@@ -50,41 +52,53 @@ Env vars:
 
 1. Set up Codespaces from the GitHub repo
 
-2. Create CosmosKey file within the secrets folder
-
-3. Input credentials in CosmosUrl and CosmosKey files within secrets folder
+2. Write CosmosDB URL in [secrets/CosmosUrl](./secrets/CosmosUrl)
 
-4. Run the application
+3. Create `CosmosKey` file in [./secrets](./secrets) folder and write the primary key
 
-```bash
-
-# run the application
-mvn clean spring-boot:run
-
-```
+   > [Optional]: You can also use Azure CLI to authenticate to CosmosDB. See [this section](#alternative-to-secret-key-cosmosdb-access-using-azure-cli)
 
-### Using bash shell
-
-> This will work from a terminal in Visual Studio Codespaces as well
+4. Run the application
+    - From terminal:
 
-1. Clone the repo
+        ```bash
+        # run the application
+        mvn clean spring-boot:run
+        ```
 
-> git clone <https://github.com/retaildevcrews/ngsa-java.git>
+    - Codespace/VSCode IDE
+       - Press F5 or select `Run/Start Debugging`
+    > wait for `Netty started on port(s): 8080`
 
-2. Create CosmosKey file within secrets folder
+### [Alternative to secret key] CosmosDB access using Azure CLI
 
-3. Input credentials in CosmosUrl and CosmosKey files within secrets folder
+We can use Azure CLI to authenticate ourselves to use CosmosDB without using Secret Key for Local Development.
 
-4. Run the application
+Follow the steps below:
 
-```bash
+1. Add your AAD user to CosmosDB (in terminal):
 
-# run the application
-mvn clean spring-boot:run
+    ```bash
+    # Login to azure
+    az login
+    # Select proper subscription/tenant using 
+    az account set -s 'SUBSCRIPTION-NAME'
+    
+    # Get your own Principal ID (replace the email with yours)
+    export PRINCIPAL=$(az ad signed-in-user show --query 'id' -o tsv)
```

Comment:
az ad command requires user be logged in on managed device. This is causing problems in the new subscription

Reply:
Debugged error using codespaces locally, and this is not an issue